### PR TITLE
Adding flaskenv, modified requirements to add more Python packages

### PR DESCRIPTION
**Pull request by LewisStaples**
Added flaskenv, modified requirements to add more Python packages

### CODE REVIEW

**Comment by dwreeves** [comment]
1. Can you fix `requirements.txt` and then redo this PR or squash the commit? The `requirements.txt` is weird (it's a binary, not a plaintext).

2. I'll probably pull this if it's ready, but in general I don't think we should do the merge thing between dev and dev_postgres piecemeal. It's a lot easier for people working on `dev_postgres` if the merge is in one fell swoop.

**Comment by dwreeves** [comment]
The other thing, other than the encoding change (not sure how that happened; is it a powershell thing?), is that there are 49 lines in the `dev` requirements.txt but there are fewer than that here. There should be at least 50 lines in the new version for it to be complete (it should contain the union of everything in dev and in dev_postgres). I'd only merge a complete requirements.txt, in addition to it being UTF-8 encoded.

**Comment by LewisStaples** [comment]
An API for boathouse metadata has been added, as described in issue # 75

File: flagging_site/data/database.py [hunk 1]
```diff
@@ -105,17 +107,22 @@ def update_database():
     model_outs.to_sql('model_outputs', **options)
 
 
+@dataclass
 class boathouses(db.Model):
```
Comment (dwreeves) [inline flagging_site/data/database.py:111]: small but important: turn this into `Boathouses` with a capital letter. This is one of the more important things to adhere to with PEP 8: all classes should start with capital letters. The exceptions to this are very rare, and only apply for function-like classes (can start with lowercase) e.g. context managers are allowed to start with lowercase.

File: flagging_site/blueprints/boathouses_api.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            type: object
+            properties:
+              boathouse:
+                type: string
+              latitude:
+                type: number
+              longitude:
+                type: number
+              reach:
+                type: integer
```
Comment (dwreeves) [inline flagging_site/blueprints/boathouses_api.yml:25]: whoa, excellent job. It's not easy getting the OpenAPI specification technically correct on the first pass, some little errors can pop up, but you nailed it here.

In the future we can add stuff like [enums](https://swagger.io/docs/specification/data-models/enums/) to represent the range of possible values for each property, but for now what you have describes the specification really well.

File: flagging_site/templates/api/index.html [hunk 3]
```diff
@@ -25,4 +25,7 @@ <h2>Available Endpoints</h2>
     <ul>
         <li><a href="{{ url_for('api.reachesapi') }}" target="_blank">Flagging Model Outputs</a></li>
     </ul>
+    <ul>
+        <li><a href="{{ url_for('api.boathousesapi') }}" target="_blank">Boathouses</a></li>
+    </ul>
```
Comment (dwreeves) [inline flagging_site/templates/api/index.html:30]: Do this instead:

```html
    <ul>
        <li><a href="{{ url_for('api.reachesapi') }}" target="_blank">Flagging Model Outputs</a></li>
        <li><a href="{{ url_for('api.boathousesapi') }}" target="_blank">Boathouses</a></li>
    </ul>
```

`<ul>` means unordered list (i.e. bullet points), so you want to include all the `<li>`'s (list items) in the single list.

**Review by dwreeves — COMMENTED** [review]
very nice! I have 2 requested changes; make sure you propagate the one change re: the Boathouse classes.

File: flagging_site/data/database.py (hunk 1 above)
Comment (LewisStaples) [reply flagging_site/data/database.py]: I have capitalized class Boathouses.

File: flagging_site/templates/api/index.html (hunk 3 above)
Comment (LewisStaples) [reply flagging_site/templates/api/index.html]: I have replaced the pair of single-item lists with a single list containing two items.

**Comment by dwreeves** [comment]
I'll be able to look at this tomorrow!

**Comment by dwreeves** [comment]
getting this error: `TypeError: Object of type Decimal is not JSON serializable` when I try your boathouses API. But I found the issue, so I'll pull this in and then I'll show you how I fixed this (Basically you need to overwrite the JSON serializer that the Flask app uses to handle Decimal types).

**Comment by dwreeves** [comment]
This issue is being merged from #95 instead as it fixes the boathouses API issue.

**Pull request closed without merging**